source/core/experimental/stdcpp/typeinfo.d

```diff
@@ -143,5 +142,144 @@ else version (CppRuntime_Gcc)
         override const(char)* what() const;
     }
 }
+else version (CppRuntime_Clang)
+{
+    extern (C++, "std"):
+
+    static assert(__traits(classInstanceSize, type_info)  == 16);
+    static assert(__traits(classInstanceSize, bad_cast)   ==  8);
+    static assert(__traits(classInstanceSize, bad_typeid) ==  8);
+
+    class type_info
+    {
+        ~this() {}
+
+    final:
+        this(const ref type_info);
+        // identity assignment operator overload is illegal
+        ref type_info opAssign_(const ref type_info);
+
+        version(LIBCPP_HAS_NONUNIQUE_TYPEINFO)
+        {
+            pragma(inline, true) extern(D)
+            int __compare_nonunique_names(const ref type_info __arg) const nothrow
```

Comment:
What's this? I can't see it in the libc++ source. Link?

Reply:
https://opensource.apple.com/source/clang/clang-800.0.42.1/src/projects/libcxx/include/typeinfo.auto.html

Reply:
Hmm, I'd rather refer to mainline LLVM: https://github.com/llvm-mirror/libcxx/blob/master/include/typeinfo
Have Apple forked it? Does apple have patches?

Reply:
To be fair, I just looked at what's installed on my machine, not the actual llvm-mirror code